core/core-services-impl/src/main/java/com/sflpro/notifier/services/notification/impl/email/EmailSenderProvider.java

```diff
@@ -0,0 +1,18 @@
+package com.sflpro.notifier.services.notification.impl.email;
+
+import com.sflpro.notifier.email.SimpleEmailSender;
+import com.sflpro.notifier.email.TemplatedEmailSender;
+
+import java.util.Optional;
+
+/**
+ * Created by Hayk Mkrtchyan.
+ * Date: 6/19/19
+ * Time: 10:24 AM
+ */
+interface EmailSenderProvider {
```

Comment:
Currently, we have to use "NotificationProcessingService" because they are
also updating the notification states during the sending process. In this case, 
having a generic sender provider interface with a generic sender provider will not give any benefits as we are using providers inside notification type specific processors.
So, I suggest,  at first, make processors generic, which could be done with different pull request.